mdal/frmts/mdal_selafin.hpp

```diff
@@ -21,20 +21,48 @@ namespace MDAL
 {
   /**
    * This class is used to read the selafin file format.
-   * The file is opened with initialize() and stay opened until this object is destructed
+   * The file is opened with initialize() and stay opened until this object is destroyed
    *
    * \note SerafinStreamReader object is shared between different datasets, with the mesh and its iterators.
    *       As SerafinStreamReader is not thread safe, it has to be shared in the same thread.
    *
    * Each record of the stream has a int (4 bytes) at the beginning and the end of the record, this int is the size of the record
   */
-  class SerafinStreamReader
+  class SelafinFile
   {
     public:
       //! Constructs the reader
-      SerafinStreamReader();
+      SelafinFile( const std::string &fileName );
+
+      //! Read the header of the file and return the project name
+      std::string readHeader();
+
+      //! Returns the vertices count in the mesh stored in the file
+      size_t verticesCount() const;
+      //! Returns the faces count in the mesh stored in the file
+      size_t facesCount() const;
+      //! Returns the vertices count per face for the mesh stored in the file
+      size_t verticesPerFace() const;
+
+      //! Creates and returns a mesh with the file
+      static std::unique_ptr<Mesh> createMesh( const std::string &fileName );
+
+      //! Method used by DatasetSelafin, returns \a count values at \a timeStepIndex and \a variableIndex, and an \a offset from the start
```

Comment:
Method used by DatasetSelafin, returns -> Returns (developer can use his favourite code editor to see where/how it is used), detto for all functions below